### PR TITLE
Register methods marked with @JsonCreator for reflection

### PR DESCRIPTION
**Pull request by gsmet**

### CODE REVIEW

**Comment by gsmet** [comment]
@geoand does it make sense to you?

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java [hunk 1]
```diff
@@ -66,6 +68,9 @@
     @Inject
     BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchyClass;
 
+    @Inject
+    BuildProducer<ReflectiveMethodBuildItem> reflectiveMethod;
```
Comment (geoand) [inline extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: Man do I dislike this pattern...

But of course I also like consistency, so better be consistently ugly :)

**Review by geoand — APPROVED** [review]
Makes perfect sense!

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java (hunk 1 above)
Comment (gsmet) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: Yeah, same here. I could change it but we have more pressing things to do, I think :).

Comment (geoand) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: Absolutely!

**Pull request merged**